### PR TITLE
[IOTDB-6259] Bump ratis version to 3.0.0

### PR DESCRIPTION
**Pull request by SzyWilliam**
Bump ratis version to 3.0.0 and change the metrics implementation.

The metrics can be correctly created and referred. see
image

### CODE REVIEW

**Review by OneSizeFitsQuorum — COMMENTED** [review]
Finally waiting for the 3.0 update! Thanks for william for working this!

Please add a manual test to prove that metrics in ratis can still be exported.

You can chain screenshots in pr~

**Comment by SzyWilliam** [comment]
> whether there are compatibility issues with the upgrade.
@wangchao316 The only incompatible change introduced in Ratis 3.0.0 is metrics upgrade. We no longer depend on dropwizard 3.x. Other things are completely compatible.

File: iotdb-core/consensus/pom.xml [hunk 1]
```diff
@@ -41,6 +41,10 @@
             <artifactId>node-commons</artifactId>
             <version>1.3.1-SNAPSHOT</version>
         </dependency>
+        <dependency>
```
[the remark this answers is not in the document]
Comment (SzyWilliam) [reply iotdb-core/consensus/pom.xml:44]: we no longer directly depend on dropwizard-core. It's introduced by iotdb-commons. Just move it nearby and wont affect any other things.

**Review by OneSizeFitsQuorum — APPROVED** [review]
LGTM

**Comment by OneSizeFitsQuorum** [comment]
Performance test：https://apache-iotdb.feishu.cn/docx/V7YydV8mYoWXBFxa1JMco1x6nVT

File: iotdb-core/consensus/src/main/java/org/apache/iotdb/consensus/ratis/metrics/IoTDBMetricRegistry.java [hunk 2]
```diff
@@ -161,93 +145,22 @@ public boolean remove(String name) {
   }
 
   @Override
-  public Gauge gauge(String name, MetricRegistry.MetricSupplier<Gauge> metricSupplier) {
-    final String fullName = getMetricName(name);
```
Comment (SzyWilliam) [inline iotdb-core/consensus/src/main/java/org/apache/iotdb/consensus/ratis/metrics/IoTDBMetricRegistry.java:165]: Wrong

**Pull request merged**